src/statement/hydrateDate.ts

```diff
@@ -0,0 +1,56 @@
+import { zeroPad } from "../common/util";
+
+// will match such formats
+// 2006-01-02
+
+// 9999-12-31 23:59:59.999999
+
+// 9999-12-31 23:59:59
+
+// 2006-1-2 1:24:5.000+00
+// 2006-1-2 1:24:5.000-07
+// 2006-1-2 15:04:05.000-07
+// 2006-01-02 15:04:05.000000-07
+// 2006-01-02 15:04:05.000000-07:00
+// 2006-01-02 15:04:05.000000-07:00:00
+// 2006-01-02 15:04:05-07
+// 2006-01-02 15:04:05-07:00
+// 2006-01-02 15:04:05-07:00:00
+
+// 2105-12-31 23:59:59
+
+// 2033-02-13 2:15:10+7:00
+
+const DATE_FORMAT =
+  /^(\d{4})-(\d{1,2})-(\d{1,2})(?:[\s+|T](\d{1,2}):(\d{1,2}):(\d{1,2})(?:\.(\d{0,6}))?(?:([+-])(\d{1,2}))?(?::(\d{2}))?(?::(\d{2}))?)?$/;
```

Comment:
Some issues with the regex here. Breaking it down with some added comments:

```
^
(\d{4})-(\d{1,2})-(\d{1,2})
(?:
    [\s+|T]            # should be (?:\s+|T), I personally wouldn't allow linebreaks \n here, but they are included in \s
    (\d{1,2}):
    (\d{1,2}):
    (\d{1,2})          # usually seconds are optional (e.g. the ISO 8601 standard)
    (?:\.(\d{0,6}))?   # this allows just a . without any digits after it, which doesn't seem valid

    (?:
        ([+-])
        (\d{1,2})
    )?                 # maybe allow the standard "Z" for UTC?
    (?::(\d{2}))?      # : are optional in ISO 8601, should be in the timezone group as it is not valid by it self
    (?::(\d{2}))?      # should be in the minute group, as it's not valid by it self nor without TZ minutes nor
)?
$
```

Adjusted:
```
^
(\d{4})-(\d{1,2})-(\d{1,2})
(?:
    (?:[ \t]+|T)
    (\d{1,2}):
    (\d{1,2})
    (?:
        :(\d{1,2})
        (?:\.(\d{1,6}))?
    )?

    (?:
        ([+-])
        (\d{1,2})
        (?:
            :?(\d{2})
            (?::?(\d{2}))?
        )?
    |
        Z
    )?
)?
$
```

```suggestion
  /^(\d{4})-(\d{1,2})-(\d{1,2})(?:(?:[ \t]+|T)(\d{1,2}):(\d{1,2})(?::(\d{1,2})(?:\.(\d{1,6}))?)?(?:([+-])(\d{1,2})(?::?(\d{2})(?::?(\d{2}))?)?|Z)?)?$/;
```